pallets/ah-migrator/src/account.rs

```diff
@@ -169,40 +169,61 @@ impl<T: Config> Pallet<T> {
 	pub fn finish_accounts_migration(rc_balance_kept: T::Balance) -> Result<(), Error<T>> {
 		use frame_support::traits::Currency;
 		let checking_account = T::CheckingAccount::get();
-		// current value is the migrated checking balance of RC
-		let rc_balance_before =
+		let balances_before = AhBalancesBefore::<T>::get();
+		// current value is the AH checking balance + migrated checking balance of RC
+		let checking_balance =
 			<<T as pallet::Config>::Currency as Currency<_>>::total_balance(&checking_account);
-		let ah_issuance_before = AhTotalIssuanceBefore::<T>::get();
 
+		/* Arithmetics explanation:
+		At this point, because checking account was completely migrated:
+			`checking_balance` = ah_check_before + rc_check_before
+			(0) rc_check_before = `checking_balance` - ah_check_before
+
+		Invariants:
+			(1) rc_check_before = sum_total_before(ah, bh, collectives, coretime, people)
+			(2) rc_check_before = sum_total_before(bh, collectives, coretime, people) + ah_total_before
+		Because teleports are disabled for RC and AH during migration, we can say:
+			(3) sum_total_before(bh, collectives, coretime, people) = sum_total_after(bh, collectives, coretime, people)
+		Ergo use (3) in (2):
+			(4) rc_check_before = sum_total_after(bh, collectives, coretime, people) + ah_total_before
+
+		We want:
+			ah_check_after = sum_total_after(rc, bh, collectives, coretime, people)
+			ah_check_after = sum_total_after(bh, collectives, coretime, people) + rc_balance_kept
+		Use (3):
+			ah_check_after = sum_total_before(bh, collectives, coretime, people) + rc_balance_kept
+			ah_check_after = sum_total_before(ah, bh, collectives, coretime, people) - ah_total_before + rc_balance_kept
+		Use (1):
+			ah_check_after = rc_check_before - ah_total_before + rc_balance_kept
+		Use (0):
+			ah_check_after = `checking_balance` - ah_check_before - ah_total_before + rc_balance_kept
+			ah_check_after = `checking_balance` + rc_balance_kept - ah_total_before - ah_check_before
+		*/
 		// set it to the correct value:
-		let balance_after = rc_balance_before
-			.checked_sub(ah_issuance_before) // amount teleported to other system chains
+		let balance_after = checking_balance
+			.checked_add(rc_balance_kept)
+			.ok_or(Error::<T>::FailedToCalculateCheckingAccount)?
+			.checked_sub(balances_before.total_issuance)
 			.ok_or(Error::<T>::FailedToCalculateCheckingAccount)?
-			.checked_add(rc_balance_kept) // total amount teleported to other chains
+			.checked_sub(balances_before.checking_account)
```

Comment:
cc @muharem

Reply:
one thing I wonder, if we should have a stage dedicated for the checking balance location switch. We could keep the teleports working for the entire migration, and disable it only for few blocks.

Reply:
No, we cannot because any teleports with other system chains will mess up the tracking of total balances on each chain. To coordinate migration with teleports enabled, we'd need to involve all system chains in the coordination.

What we could do however is reduce the teleports disabled window from `[MigrationStart, MigrationEnd]` to `[AccountsMigrationStart, AccountsMigrationEnd]`.